src/main/java/jimmy/mcgymmy/model/McGymmy.java

```diff
@@ -0,0 +1,121 @@
+package jimmy.mcgymmy.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import jimmy.mcgymmy.model.food.Food;
+import jimmy.mcgymmy.model.food.Fridge;
+import jimmy.mcgymmy.model.food.exceptions.DuplicateFoodException;
+
+/**
+ * Wraps all data at mcgymmy level
+ * Duplicates are not allowed (by .isSameFood comparison)
+ */
+public class McGymmy implements ReadOnlyMcGymmy {
+
+    private final Fridge foodItems;
+
+    public McGymmy() {
+        foodItems = new Fridge();
+    }
+
+    /**
+     * Creates an McGymmy using the Foods in the {@code toBeCopied}
+     */
+    public McGymmy(ReadOnlyMcGymmy toBeCopied) {
+        this();
+        resetData(toBeCopied);
+    }
+
+    // list overwrite operations
+
+    /**
+     * Replaces the contents of the Food list with {@code Foods}.
+     * {@code Foods} must not contain duplicate Foods.
+     */
+    public void setFoodItems(List<Food> foodItems) {
+        this.foodItems.setFoods(foodItems);
+    }
+
+    /**
+     * Resets the existing data of this {@code McGymmy} with {@code newData}.
+     */
+    public void resetData(ReadOnlyMcGymmy newData) {
+        requireNonNull(newData);
+        for (Food food : newData.getFoodList()) {
+            for (Food comp : newData.getFoodList()) {
+                if (food.hashCode() == comp.hashCode()) {
```

Comment:
this should be `if (food == comp)` for referential equality (the `is` operator in python).

Comparing the hashcodes does not guarantee the food and comp are the same object in memory.